### PR TITLE
[docs] Update consensus

### PR DESCRIPTION
**Pull request by ronny-mysten**
## Description 

Adds some info to the consensus/mysticeti page

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Review by tedks — APPROVED** [review]
LGTM!

File: docs/content/concepts/sui-architecture/consensus.mdx [hunk 1]
```diff
@@ -8,11 +8,34 @@ The basic purpose of consensus in blockchains is to agree on a consistent order
...
+
+For more details, including correctness proofs, see the [MYSTICETI: Reaching the Latency Limits with
+Uncertified DAGs](https://arxiv.org/pdf/2310.14821) whitepaper.
+
+## Related links
+
+- [Life of a Transaction](./transaction-lifecycle.mdx): The life of a transaction on the Sui network has some differences compared to those from other blockchains.
+- [Sui Security](./sui-security.mdx): Sui is designed to provide very high security guarantees to asset owners.
+- [MYSTICETI: Reaching the Latency Limits with Uncertified DAGs](https://arxiv.org/pdf/2310.14821): Whitepaper documenting the Mysticeti protocol.
```
Comment (tedks) [inline docs/content/concepts/sui-architecture/consensus.mdx:41]: Do we have another link besides arxiv we could add here? @mwtian

Comment (ronny-mysten) [reply docs/content/concepts/sui-architecture/consensus.mdx:41]: I'm pretty sure we can just host the pdf. I'll check to be sure.

File: docs/content/concepts/sui-architecture/consensus.mdx [hunk 2]
```diff
@@ -8,11 +8,34 @@ The basic purpose of consensus in blockchains is to agree on a consistent order
...
 
-Byzantine-fault tolerant (BFT) consensus protocols are a rich area of research. The next-generation consensus engine set to roll out in Sui is based on the **Mysticeti** protocol.
-The protocol optimizes for both low latency and high throughput because:
-- It allows multiple validators to propose blocks in parallel, utilizing the full bandwidth of the network and providing censorship resistance. These are features of the DAG-based consensus protocols.
-- It takes only three rounds of messages to commit blocks from the DAGs, same as pBFT and matches the theoretical minimum.
+Sui uses the Mysticeti protocol to optimize for both low latency and high throughput. The benefits of Mysticeti include:
+
+- Supports multiple validators proposing blocks in parallel, utilizing the full bandwidth of the network and providing censorship resistance. These are features of the DAG-based consensus protocols.
+- The first consensus protocol to require only three rounds of messages to commit blocks from the DAGs (first of its kind), same as practical Byzantine Fault Tolerance and matches the theoretical minimum.
```
Comment (akichidis) [inline docs/content/concepts/sui-architecture/consensus.mdx:14]: CC @asonnino . Is this safe to be claimed here? I mean as the "first" one?

File: docs/content/concepts/sui-architecture/consensus.mdx [hunk 3]
```diff
@@ -8,11 +8,34 @@ The basic purpose of consensus in blockchains is to agree on a consistent order
...
+
+## Finality
+
+Finality is the guarantee that a transaction or block, after confirmation, is permanently added to the network and cannot be altered or reversed. In traditional blockchain consensus, confirming transactions can take time because they rely on other transactions to “reference” them repeatedly before they are considered final. This process can slow down if the network activity decreases or if there are many competing transactions. In contrast, Mysticeti simplifies this process by finalizing transactions immediately upon inclusion in the structure. Consequently, there's no waiting for additional confirmations or network activity, making Mysticeti faster and more reliable for confirming transactions, even in less active or challenging network conditions.
```
Comment (akichidis) [inline docs/content/concepts/sui-architecture/consensus.mdx:32]: My understanding is here that we are comparing Mysticeti to chains mostly with non BFT chains like Bitcoin etc base safety on confirmations (well I guess some BFT consensus systems also use similar concepts) . If that's the intention here then that looks good to me. If we want to make more direct comparison with other state of the art consensus engines then we might want to be more specific. That being said, depending on the intention to remain more high level that's fine.

Comment (ronny-mysten) [reply docs/content/concepts/sui-architecture/consensus.mdx:32]: That was the original intention. The idea is that this topic remains high and anyone wanting more details can reference the whitepaper.

File: docs/content/concepts/sui-architecture/consensus.mdx (hunk 2 above)
Comment (ronny-mysten) [reply docs/content/concepts/sui-architecture/consensus.mdx]: I think I got this from the whitepaper, but it's really a no-value statement here so removing.

**Pull request merged**